assets/js/main.js

```diff
@@ -1,29 +1,26 @@
-$(window).scroll(function() {
-  
-    // selectors
-    var $window = $(window),
-        $body = $('body'),
-        $panel = $('.scroll-bg');
-    
-    // Change 33% earlier than scroll position so colour is there when you arrive.
-    var scroll = $window.scrollTop() + ($window.height() / 3);
-   
-    $panel.each(function () {
-      var $this = $(this);
-      
-      // if position is within range of this panel.
-      // So position of (position of top of div <= scroll position) && (position of bottom of div > scroll position).
-      // Remember we set the scroll to 33% earlier in scroll var.
-      if ($this.position().top <= scroll && $this.position().top + $this.height() > scroll) {
-            
-        // Remove all classes on body with color-
-        $body.removeClass(function (index, css) {
-          return (css.match (/(^|\s)color-\S+/g) || []).join(' ');
-        });
-         
-        // Add class of currently active div
-        $body.addClass('color-' + $(this).data('color'));
-      }
-    });    
-    
-  }).scroll();
+window.addEventListener("scroll", function() {
+  // selectors
+  var body = document.body,
+      panels = document.querySelectorAll('.scroll-bg');
+
+  // Iterate through each panel
+  panels.forEach(function (panel) {
+    var rect = panel.getBoundingClientRect();
+
+    // Calculate the top and bottom scroll positions for this panel
+    var panelTop = rect.top + window.pageYOffset;
+    var panelBottom = panelTop + rect.height;
+
+    // Check if the current scroll position is within the panel's range
+    if (window.pageYOffset >= panelTop && window.pageYOffset < panelBottom) {
+      // Remove all classes on body with color-
+      body.className = body.className.replace(/\bcolor-\S+/g, '');
+
+      // Add class of the currently active div
+      body.classList.add('color-' + panel.getAttribute('data-color'));
+    }
+  });
+});
+
+// Trigger the scroll event once to apply the initial state
+window.dispatchEvent(new Event('scroll'));
```

Comment:
@farski I ran the initial jQuery through a converter to vanilla JS and seem to have it working as expected (with an assist by ChatGPT. I know, I know.)